js/jquery.validation.js

```diff
@@ -0,0 +1,222 @@
+                    type: 'POST',
+                    dataType: 'JSON',
+                    success: function(response, success, request) {
+                        var allValid = true;
+                        // Check each record item is valid.
+                        $.each(response.record, function (name, object) {
+                            var $element = $form.find('[name=' + name + ']');
+                            if ( ! object.valid ) {
+                                // Record item not valid - mark invalid and display message to user.
+                                allValid = false;
+                                if ( $element.length !== 0 ) {
+                                    $form.validation('showValidationMessage', $element, object.message);
+                                    $element.addClass('invalid');
+                                }
+                            } else {
+                                $element.removeClass('invalid');
+                            }
+                        });
+                        
+                        if ( allValid && response.status === 'valid') {
+                            if (response.action && response.action['redirect']) {
+                                // Redirect if record was valid and the redirect action was given
+                                window.location.href = response.action['redirect'].value;
+                            } else if (response.message['notify']) {
+                                $form.validation('showMessage', 'notify', response.message['notify'].value);
+                                $form.validation('reposition');
+                            }
+                            // Reset the form
+                            $form.trigger('reset');
+                            
+                        } else if (response.message) {
+                            $.each(response.message, function(type, object) {
+                                $form.validation('showMessage', type, object.value);
+                                $form.validation('reposition');
+                            });
+                        }
+                        // Trigger validation completion event
+                        $form.trigger({
+                            type: 'validationComplete',
+                            response: response
+                        });
+                    },
+                    error: function(response, error, message) {
+                        var myWindow = window.open('', '_parent');
+                        myWindow.document.write(response.responseText);
+                    }
+                });
+            }
+            
+            $(this.element).on('submit.validate', validate);
+        },
+        
+        showValidationMessage: function($element, message) {
+            // Show the validation message with the message as the content for the given element.
```

Comment:
There's `showValidationMessage` and `showMessage`.

`showValidationMessage` displays the qtips on an element while `showMessage` displays the `alert`, `notify`, or `error` messages.

I feel that one (or maybe both?) should be renamed as there could be confusion with the names as they currently stand.

Some ideas:

`showValidationMessage`
  * `showQtip`
  * `showHint`
  * `showRecordItemMessage`

Any better names for either `showValidationMessage` or `showMessage`?